### PR TITLE
add description of system query endpoint to dataservice documentation

### PR DESCRIPTION
**Pull request by jusong**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/data-service-manage-endpoint.md [hunk 1]
```diff
@@ -257,6 +295,10 @@ To rename an endpoint, perform the following steps:
 2. In the left pane, click the name of your target Data App to view its endpoints.
 3. Locate the endpoint you want to rename, click **...** > **Rename**., and enter a new name for the endpoint.
 
+> **Note:**
+>
+> Sstem-provided endpoints do not support renaming.
```
Comment (hfxsd) [inline tidb-cloud/data-service-manage-endpoint.md:300]:
```suggestion
> System-provided endpoints do not support renaming.
```

File: tidb-cloud/data-service-manage-endpoint.md [hunk 2]
```diff
@@ -73,6 +73,43 @@ To create an endpoint manually, perform the following steps:
 3. Update the default name if necessary. The newly created endpoint is added to the top of the endpoint list.
 4. Configure the new endpoint according to the instructions in [Develop an endpoint](#develop-an-endpoint).
 
+### Add a predefined system endpoint
+
+Data Service provides an endpoint library with a list of predefined system endpoints to reduce your effort of endpoint creation. Currently, the library only includes the `/system/query` endpoint, which offers the capability to execute any SQL statements.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-endpoint.md:78]:
```suggestion
Data Service provides an endpoint library with predefined system endpoints that you can directly add to your Data App, reducing the effort in your endpoint development. Currently, the library only includes the `/system/query` endpoint, which offers the capability to execute any SQL statements.
```

File: tidb-cloud/data-service-manage-endpoint.md [hunk 3]
```diff
@@ -73,6 +73,43 @@ To create an endpoint manually, perform the following steps:
 3. Update the default name if necessary. The newly created endpoint is added to the top of the endpoint list.
 4. Configure the new endpoint according to the instructions in [Develop an endpoint](#develop-an-endpoint).
 
+### Add a predefined system endpoint
+
+Data Service provides an endpoint library with predefined system endpoints that you can directly add to your Data App, reducing the effort in your endpoint development. Currently, the library only includes the `/system/query` endpoint, which offers the capability to execute any SQL statements.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-endpoint.md:78]:
```suggestion
Data Service provides an endpoint library with predefined system endpoints that you can directly add to your Data App, reducing the effort in your endpoint development. Currently, the library only includes the `/system/query` endpoint, which enables you to execute any SQL statement by simply passing the statement in the predefined `sql` parameter.
```

File: tidb-cloud/data-service-app-config-files.md [hunk 4]
```diff
@@ -187,7 +187,7 @@ The description of each field is as follows:
 | `tag`    | String | The tag for the endpoint. The default value is `"Default"`. |
 | `batch_operation`    | Integer | Controls whether to enable the endpoint to operate in batch mode. Supported values are `0` (disabled) and `1` (enabled). When it is set to `1`, you can operate on multiple rows in a single request. To enable this option, make sure that the request method is `POST` or `PUT`. |
 | `sql_file`    | String | The SQL file directory for the endpoint. For example, `"sql/GET-v1.sql"`. |
-| `type`        | String | The type of the endpoint, which can only be `"sql_endpoint"`.          |
+| `type`        | String | The type of the endpoint. The value is `"system-data"` for predefined system endpoints and  `"sql_endpoint"` for other endpoints. |
```
Comment (hfxsd) [inline tidb-cloud/data-service-app-config-files.md:190]:
```suggestion
| `type`        | String | The type of the endpoint. The value is `"system-data"` for predefined system endpoints and `"sql_endpoint"` for other endpoints. |
```

File: tidb-cloud/data-service-manage-endpoint.md [hunk 5]
```diff
@@ -73,6 +73,43 @@ To create an endpoint manually, perform the following steps:
 3. Update the default name if necessary. The newly created endpoint is added to the top of the endpoint list.
 4. Configure the new endpoint according to the instructions in [Develop an endpoint](#develop-an-endpoint).
 
+### Add a predefined system endpoint
+
+Data Service provides an endpoint library with predefined system endpoints that you can directly add to your Data App, reducing the effort in your endpoint development. Currently, the library only includes the `/system/query` endpoint, which enables you to execute any SQL statement by simply passing the statement in the predefined `sql` parameter.
+
+To add a predefined system endpoint to your Data App, perform the following steps:
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project.
```
Comment (hfxsd) [inline tidb-cloud/data-service-manage-endpoint.md:82]:
```suggestion
1. Navigate to the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project.

```

File: tidb-cloud/data-service-manage-endpoint.md [hunk 6]
```diff
@@ -73,6 +73,43 @@ To create an endpoint manually, perform the following steps:
...
+    The added system-provided endpoint is displayed at the top of the endpoint list.
+
+5. Check the endpoint name, SQL statements, properties, and parameters of the new endpoint.
+
+    > **Caution:**
+    >
+    > The `/system/query` endpoint is powerful and versatile but can be potentially destructive. Use it with discretion and ensure the queries are secure and well-considered to prevent unintended consequences.
+
+    - Endpoint name: the endpoint name and path is `/system/query`, and the request method `POST`.
+    - SQL statements: the `/system/query` endpoint does not come with any SQL statement. You can write your desired SQL statements in the middle section of the page. Note that SQL statements for the `/system/query` endpoint are for console development and testing only and will not be saved.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-endpoint.md:109]:
```suggestion
    - SQL statements: the `/system/query` endpoint does not come with any SQL statement. You can write your desired SQL statements in the SQL editor of the page. Note that SQL statements written in the SQL editor for the `/system/query` endpoint are for development and testing only and will not be saved in the endpoint configuration.
```

File: tidb-cloud/data-service-manage-endpoint.md [hunk 7]
```diff
@@ -73,6 +73,43 @@ To create an endpoint manually, perform the following steps:
...
+    > **Tip:**
+    >
+    > To remove an added predefined endpoint from your Data App, toggle the **Execute Query** switch to **Removed**.
+
+4. Click **Save**.
+
+    > **Note:**
+    >
+    > - After you click **Save**, the added or removed endpoint is deployed to production immediately, which makes the added endpoint accessible and the removed endpoint inaccessible immediately.
+    > - If a non-predefined endpoint with the same path and method already exists in the current app, the creation of the system endpoint will fail.
```
Comment (hfxsd) [inline tidb-cloud/data-service-manage-endpoint.md:98]:
```suggestion
    > - If a non-predefined endpoint with the same path and method already exists in the current App, the creation of the system endpoint will fail.
```

File: tidb-cloud/data-service-manage-endpoint.md (hunk 6 above)
Comment (qiancai) [reply tidb-cloud/data-service-manage-endpoint.md:109]: @jusong PTAL

File: tidb-cloud/data-service-manage-endpoint.md [hunk 8]
```diff
@@ -73,6 +73,43 @@ To create an endpoint manually, perform the following steps:
...
+5. Check the endpoint name, SQL statements, properties, and parameters of the new endpoint.
+
+    > **Caution:**
+    >
+    > The `/system/query` endpoint is powerful and versatile but can be potentially destructive. Use it with discretion and ensure the queries are secure and well-considered to prevent unintended consequences.
+
+    - Endpoint name: the endpoint name and path is `/system/query`, and the request method `POST`.
+    - SQL statements: the `/system/query` endpoint does not come with any SQL statement. You can write your desired SQL statements in the middle section of the page. Note that SQL statements for the `/system/query` endpoint are for console development and testing only and will not be saved.
+    - Endpoint properties: unlike other custom endpoints, only the `timeout` and `max rows` properties can be customized for system endpoints.
+    - Endpoint parameters: the endpoint parameters are configured automatically and cannot be modified.
```
Comment (hfxsd) [inline tidb-cloud/data-service-manage-endpoint.md:111]:
```suggestion
    - **Endpoint name**: the endpoint name and path is `/system/query`, and the request method `POST`.
    - **SQL statements**: the `/system/query` endpoint does not come with any SQL statement. You can write your desired SQL statements in the middle section of the page. Note that SQL statements for the `/system/query` endpoint are for console development and testing only, and will not be saved.
    - **Endpoint properties**: unlike other custom endpoints, only the `timeout` and `max rows` properties can be customized for system endpoints.
    - **Endpoint parameters**: the endpoint parameters are configured automatically and cannot be modified.
```

Comment (qiancai) [inline tidb-cloud/data-service-manage-endpoint.md:111]:
```suggestion
    - Endpoint properties: you can find the properties in the **Properties** tab of the right pane of the page. unlike other custom endpoints, only the `timeout` and `max rows` properties can be customized for system endpoints. 
    - Endpoint parameters: you can find the parameters in the **Params** tab of the right pane of the page. The parameters of the `/system/query` endpoint are configured automatically and cannot be modified. 
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**